lib/check-meta.nix

```diff
@@ -0,0 +1,200 @@
+let
+  lib = import ./default.nix;
+in rec {
+
+  # Extend a derivation with checks for brokenness, license, etc.
+  # Throw a descriptive error when the check fails; return derivaitonArg otherwise.
+  # Note: no dependencies are checked in this step.
+  addMetaChecks = { config, platform, meta, derivationArg, mkDerivationArg }: let
```

Comment:
This should be `system` not `platform`, because while those terms are almost use interchangeably in nixpkgs, the two-part dash-separated string name is always called `system`.

In the original code is *was* called system, so I that's why I bring this up now---once I fix for cross, it will all be gone anyways.